src/commons/asBaseComponent.tsx

```diff
@@ -23,11 +24,22 @@ function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentT
     static displayName: string | undefined;
     static propTypes: any;
     static defaultProps: any;
+    appearanceListener: any;
```

Comment:
```suggestion
    appearanceListener: appearanceListener?: Appearance.AppearanceListener;
```
Once this is changed you'll see the need for the other two changes in this file.

Reply:
After thinking about it, this is better:
```
appearanceListener = ({colorScheme}: {colorScheme: ColorSchemeName}) => {
  this.setState({colorScheme});
};
```
This will remove the need for the `if` in `componentWillUnmount`